pasqal_cloud/workload.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, Extra
+
+from pasqal_cloud.client import Client
+
+
+class Workload(BaseModel):
+    """Class for workload data.
+
+    Attributes:
+        - id: Unique identifier for the workload.
+        - project_id: ID of the project which the users scheduling the workload belong to.
+        - status: Status of the workload. Possible values are:
+            PENDING, RUNNING, DONE, CANCELED, TIMED_OUT, ERROR, PAUSED.
+        - _client: A Client instance to connect to PCS.
+        - backend: The backend used for the workload.
+        - workload_type: The type of the workload.
+        - config: The config containing all the necessary information for the workload to run.
+        - created_at: Timestamp of the creation of the workload.
+        - updated_at: Timestamp of the last update of the workload.
+        - errors: Error messages that occurred while processing workload.
+        - start_timestamp: The timestamp of when the workload began processing.
+        - end_timestamp: The timestamp of when the workload finished processing.
+        - result: Result of the workload.
+    """
+
+    id: str
+    project_id: str
+    status: str
+    _client: Client
+    backend: Optional[str] = None
+    workload_type: Optional[str] = None
+    config: Optional[Dict[str, Any]] = None
```

Comment:
Why define a Config class below if you're going to allow an optional dict here?

Reply:
Not sure to fully understand to get your point here as the Config class is just a way to configure pydantic's behavior & is not linked to the config field of the model ? By the way as suggested above this is not Optional anymore as it didn't make sense.

Reply:
Yeah, brainfart on my side